src/java.base/share/classes/jdk/internal/classfile/instruction/CharacterRange.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.classfile.instruction;
+
+import jdk.internal.classfile.Classfile;
+import jdk.internal.classfile.CodeElement;
+import jdk.internal.classfile.CodeModel;
+import jdk.internal.classfile.Label;
+import jdk.internal.classfile.PseudoInstruction;
+import jdk.internal.classfile.attribute.CharacterRangeTableAttribute;
+import jdk.internal.classfile.impl.AbstractPseudoInstruction;
+import jdk.internal.classfile.impl.BoundCharacterRange;
+
+/**
+ * A pseudo-instruction which models a single entry in the
+ * {@link CharacterRangeTableAttribute}.  Delivered as a {@link CodeElement}
+ * during traversal of the elements of a {@link CodeModel}, according to
+ * the setting of the {@link Classfile.Option#processDebug(boolean)} option.
+ */
+public sealed interface CharacterRange extends PseudoInstruction
```

Comment:
This and some other instructions with unbounded implementations do not have static `of` factory methods. Is that intended?

Reply:
It seems to be incomplete. I'll add factory methods to `CharacterRange`, `LineNumber`, `LocalVariable` and `LocalVariableType`.
Thanks for pointing it out.

Reply:
Fixed, thanks.